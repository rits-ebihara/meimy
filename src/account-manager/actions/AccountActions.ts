import { Alert } from 'react-native';
import { getGenericPassword, setGenericPassword } from 'react-native-keychain';
import { Action, Dispatch } from 'redux';
import ShortId from 'shortid';

import { dateParser } from '../../eim-service/EIMServiceAdapter';
import { getConfig } from '../Config';
import { createInitAccountListState, IAccountListState } from '../states/IAccountLisState';
import { IAccountState } from '../states/IAccountState';
import { asyncLoadAccountListAfterShow } from './AccountListActions';
import { langProfile } from '../../LangProfile';

export const SET_ACCOUNT_ACTION = ShortId();

export interface ISetAccountAction extends Action {
    account: IAccountState;
}

export const createSetAccountAction = (account: IAccountState): ISetAccountAction => {
    return {
        account,
        type: SET_ACCOUNT_ACTION,
    };
};

const config = getConfig();

export const asyncSaveAccountAction
    = async (account: IAccountState, dispatch: Dispatch) => {
        try {
            const { siteName, siteDomain, authType, id, eimToken, lastConnect, userId, password } = account;
            // 拡張した他のプロパティが入らないように精査する
            const saveData: IAccountState = {
                authType,
                eimToken,
                id,
                lastConnect,
                password,
                siteDomain,
                siteName,
                userId,
            };
            // 現在のリストをロード
            const json = await getGenericPassword({ service: config.accountListServiceName });
            let accountList: IAccountListState = createInitAccountListState();
            if (!!json && typeof json !== 'boolean') {
                accountList = JSON.parse(json.password, dateParser) as IAccountListState;
            }
            // IDで検索しなければ追加する
            const existIndex = accountList.accounts.findIndex((a) => a.id === id);
            if (existIndex === -1) {
                accountList.accounts.push(saveData);
            } else {
                accountList.accounts[existIndex] = saveData;
            }
            await setGenericPassword('dummy', JSON.stringify(accountList), { service: config.accountListServiceName });
        } catch (error) {
            Alert.alert('error', langProfile.replaceLang('LK_MSG_saveError'));
            console.error(error);
        }
        // ステートを更新して、一覧に戻ったときに画面の表示が新しくなっているようにする
        await asyncLoadAccountListAfterShow(dispatch);
    };

export const REMOVE_ACCOUNT_ACTION = ShortId();

// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface IRemoveAccountAction extends Action {
}

export const asyncRemoveAccountAction = async (
    targetId: string, dispatch: Dispatch, onSuccess: () => void) => {
    const save = async () => {
        try {
            // 現在のリストをロード
            const json = await getGenericPassword({ service: config.accountListServiceName });
            let accountList: IAccountListState = createInitAccountListState();
            if (!!json && typeof json !== 'boolean') {
                accountList = JSON.parse(json.password) as IAccountListState;
            } else {
                return;
            }
            accountList.accounts = accountList.accounts.filter((a) => a.id !== targetId);
            await setGenericPassword('dummy', JSON.stringify(accountList), { service: config.accountListServiceName });
        } catch (error) {
            Alert.alert('error', langProfile.replaceLang('LK_MSG_saveError'));
            console.error(error);
        }
    };
    await save();
    // ステートを更新して、一覧に戻ったときに画面の表示が新しくなっているようにする
    await asyncLoadAccountListAfterShow(dispatch);
    onSuccess();
};
