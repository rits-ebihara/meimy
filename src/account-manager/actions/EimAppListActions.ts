// import { Toast } from 'native-base';
import { NavigationScreenProp } from 'react-navigation';
import { Action, Dispatch } from 'redux';
import ShortId from 'shortid';

import { EIMServiceAdapter } from '../../eim-service/EIMServiceAdapter';
import { IEimApp } from '../states/IEimAppListState';
import INavigateController from './INavigateController';

interface ISourceEimApp {
    system: {};
    properties: {
        id: string;
        name: string;
        description: string;
    };
}

interface IGetAppListResponse {
    apps: ISourceEimApp[];
}

export const SET_APP_LIST = ShortId();
export interface ISetAppListAction extends Action {
    appList: IEimApp[];
}
export const createSetAppListAction = (appList: IEimApp[]): ISetAppListAction => {
    return {
        appList,
        type: SET_APP_LIST,
    };
};

export const LOAD_APP_LIST = ShortId();

// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface ILoadAppListAction extends Action {
}
export const createLoadAppListAction =
    async (dispatch: Dispatch, navigateActions: INavigateController,
        // eslint-disable-next-line @typescript-eslint/no-explicit-any
        navigation: NavigationScreenProp<any>, onError: () => void): Promise<void> => {
        const { siteDomain, siteName, tokens, appKeyPrefix } = navigateActions.getLinkState();
        if (!siteDomain || !tokens || !appKeyPrefix) { return; }
        dispatch({
            type: LOAD_APP_LIST,
        });
        // アプリ一覧の取得
        await (async () => {
            const sa = new EIMServiceAdapter(siteDomain);
            try {
                const response = await sa.getAppList(tokens);
                const responseObj = JSON.parse(response.body) as IGetAppListResponse;
                const appList = responseObj.apps
                    .filter((a) => a.properties.id.startsWith(appKeyPrefix))
                    .map((sourceApp): IEimApp => {
                        const { properties } = sourceApp;
                        return {
                            appKey: properties.id,
                            appName: properties.name,
                            description: properties.description,
                            siteDomain,
                            siteName: siteName || siteDomain,
                            tokens,
                        };
                    });
                // リストに１件しかない場合
                if (appList.length === 1) {
                    const appKey = appList[0].appKey;
                    navigateActions.openApp({ appKey }, navigation);
                } else {
                    dispatch(createSetAppListAction(appList));
                }
            } catch (e) {
                console.error(e);
                onError();
            }
        })();
    };
