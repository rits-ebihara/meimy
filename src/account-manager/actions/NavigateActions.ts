import Clone from 'clone';
import { Linking } from 'react-native';
import { NavigationScreenProp } from 'react-navigation';
import { Dispatch } from 'redux';
import UrlParse from 'url-parse';

import { EIMServiceAdapter } from '../../eim-service/EIMServiceAdapter';
import { getConfig } from '../Config';
import { getEimAccount } from '../EimAccount';
import { IAuthAppQuery } from '../IEimAccount';
import RoutePageNames from '../RoutePageNames';
import { IAccountListState } from '../states/IAccountLisState';
import { createInitAccountState } from '../states/IAccountState';
import { IAuthState } from '../states/IAuthStates';
import { createSetAccountAction } from './AccountActions';
import { createSetAuthState } from './AccountListActions';
import { createSetAppListAction } from './EimAppListActions';
import INavigateController from './INavigateController';

export class NavigateController implements INavigateController {
    public parentMainPage?: string;
    public parentNavParams?: { [key: string]: any };
    private linkStates: IAuthState = {};
    private transferAccountPage = (
        accountListState: IAccountListState,
        dispatch: Dispatch,
        navigation: NavigationScreenProp<any>,
        replace: boolean,
        siteDomain: string,
    ) => {
        const accountList = accountListState.accounts;
        let account = accountList.find((a) => a.siteDomain === this.linkStates.siteDomain);
        if (!account) {
            // なければアカウント新規作成画面に遷移する
            account = createInitAccountState();
            const name = siteDomain;
            account.siteName = name;
            account.siteDomain = name;
        }
        dispatch(createSetAccountAction(account));
        if (replace) {
            navigation.replace(RoutePageNames.accountPageName);
        } else {
            navigation.navigate(RoutePageNames.accountPageName);
        }
    };
    private determinedAppAndDomain = async (
        siteDomain: string,
        navigation: NavigationScreenProp<any>,
        accountListState: IAccountListState,
        dispatch: Dispatch,
        replace: boolean,
    ): Promise<boolean> => {
        // トークンがある
        const sa = new EIMServiceAdapter(siteDomain);
        const connected = await sa.validateToken(this.linkStates.tokens || []);
        // トークン検証
        if (!!connected) {
            // 成功
            // 呼び出しアプリを起動
            await this.openApp({}, navigation);
            return true;
        } else {
            // 失敗
            // アカウント画面に遷移する
            this.transferAccountPage(
                accountListState, dispatch, navigation, replace, siteDomain);
            return true;
        }
    }
    private deteminedAppOnly = async (
        domain: string,
        replace: boolean,
        navigation: NavigationScreenProp<any>,
        dispatch: Dispatch,
        accountListState: IAccountListState,
    ) => {
        const sa = new EIMServiceAdapter(domain);
        // 接続済みの場合
        // トークンがある場合は、検証する
        const connected = await sa.validateToken(this.linkStates.tokens || []);
        // 検証の結果
        if (connected) {
            // 認証が成功の場合
            // アプリ一覧を取得
            // アプリ一覧の画面に遷移する
            const moveTo = (replace) ? navigation.replace : navigation.navigate;
            if (this.linkStates.appKeyPrefix) {
                dispatch(createSetAppListAction([]));
                moveTo(RoutePageNames.appListPageName);
            } else {
                moveTo(RoutePageNames.accountListPageName);
            }
            return true;
        } else {
            // 失敗の場合、アカウントの画面に遷移する
            this.transferAccountPage(
                accountListState, dispatch, navigation, replace, domain);
        }
        return true;
    }
    public navigateForLink = async (
        accountListState: IAccountListState,
        pLinkState: IAuthState,
        dispatch: Dispatch,
        // eslint-disable-next-line @typescript-eslint/no-explicit-any
        navigation: NavigationScreenProp<any>,
        replace: boolean = false
    ): Promise<boolean> => {
        this.linkStates = Object.assign({}, this.linkStates, pLinkState);
        if (!!this.linkStates.siteDomain && !!this.linkStates.appKey) {
            // ドメインとアプリが決まっている
            return await this.determinedAppAndDomain(
                this.linkStates.siteDomain,
                navigation,
                accountListState,
                dispatch,
                replace);
        } else if (!!this.linkStates.siteDomain) {
            // サイトドメインが決まっている
            // トークンがある
            return await this.deteminedAppOnly(
                this.linkStates.siteDomain,
                replace,
                navigation,
                dispatch,
                accountListState,
            );
        } else {
            // ドメインの指定がない場合は、なにもしない
            return false;
        }
    }
    public clear = () => {
        this.linkStates = {};
    }
    public getLinkState = () => {
        return Clone(this.linkStates);
    }
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    public openApp = async (pLinkState: IAuthState, navigation: NavigationScreenProp<any>) => {
        this.linkStates = Object.assign({}, this.linkStates, pLinkState);
        const eimAccount = getEimAccount();
        eimAccount.appKey = this.linkStates.appKey || '';
        eimAccount.domain = this.linkStates.siteDomain || '';
        eimAccount.siteName = this.linkStates.siteName || '';
        eimAccount.eimTokens = this.linkStates.tokens || [];
        await eimAccount.save();
        await eimAccount.loadUser();
        if (this.parentMainPage) {
            navigation.navigate(this.parentMainPage, this.parentNavParams);
        }
    }
    public openAccountManager = async (
        // eslint-disable-next-line @typescript-eslint/no-explicit-any
        navigation: NavigationScreenProp<any>,
        dispatch: Dispatch,
        link?: string,
        hash?: string
    ) => {
        let appKey: string | undefined;
        let domain: string | undefined;
        // 部分適用
        const open = (appKey?: string, domain?: string) => {
            return this._openAccountManager(
                dispatch, navigation, link, hash,
                appKey, domain);
        }
        // メールなど文書を開くリンクから開く
        if (!!link && !!hash) {
            const linkUrl = UrlParse(link);
            domain = linkUrl.host;
            const paths = hash.split('/');
            appKey = paths.length === 5 ? paths[2] : undefined;
            open(appKey, domain);
            return;
        }
        const eimAccount = getEimAccount();
        // 前回アクセスしたサイトを取得する
        await eimAccount.load().then((lastAccount) => {
            // 認証アプリを起動する
            if (!!lastAccount) {
                open(lastAccount.appKey, lastAccount.domain);
            } else {
                open();
            }
        });
    }
    private _openAccountManager = async (
        dispatch: Dispatch, navigation: NavigationScreenProp<any>,
        link?: string, hash?: string,
        appKey?: string, domain?: string
    ) => {
        const config = getConfig();

        const authState: IAuthState = {
            appKeyPrefix: config.appKeyPrefix,
        };
        const accountManagerUrl = UrlParse('eimapplink-accountmanager://accountmanager/');
        const query: IAuthAppQuery = {
            appprefix: config.appKeyPrefix,
            mapp: config.appKeyPrefix,
        };
        if (!!appKey && !!domain) {
            query.appkey = appKey;
            query.domain = domain;
            authState.appKey = appKey;
            authState.siteDomain = domain;
        }
        if (!!link) {
            const linkUrl = UrlParse(link);
            linkUrl.set('hash', hash);
            query.link = linkUrl.href;
            authState.link = linkUrl.href;
        }

        accountManagerUrl.set('query', query);
        await Linking.canOpenURL(accountManagerUrl.href).then((result) => {
            if (result) {
                Linking.openURL(accountManagerUrl.href);
            } else {
                this.parentMainPage = config.startPage;
                if (!!hash) {
                    const paths = hash.split('/');
                    this.parentNavParams = {
                        parameter: paths.length === 5 ? paths[4] : undefined,
                    };
                }
                dispatch(createSetAuthState(authState));
                navigation.navigate(RoutePageNames.authPageName);
            }
        });
    }
}
export default new NavigateController();
