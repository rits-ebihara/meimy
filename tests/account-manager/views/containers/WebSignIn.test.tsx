/* eslint-disable @typescript-eslint/no-non-null-assertion */
import Enzyme, { shallow } from 'enzyme';
import EnzymeAdapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import { Toast } from 'native-base';
import React from 'react';
import Cookies from 'react-native-cookies';
import { mocked } from 'ts-jest/utils';

import { asyncSaveAccountAction } from '../../../../src/account-manager/actions/AccountActions';
import { IAccountState } from '../../../../src/account-manager/states/IAccountState';
import {
    __private__,
    _WebSignIn as WebSignIn,
    ILocalState,
    ThisProps,
} from '../../../../src/account-manager/views/containers/WebSignIn';

Enzyme.configure({ adapter: new EnzymeAdapter() });

// const config = getConfig();

jest.mock('native-base', () => ({
    ...jest.requireActual('native-base'),
    Toast: {
        show: jest.fn().mockReturnValue({}),
    },
}));

jest.mock('react-native-cookies', () => ({
    clearAll: jest.fn(),
    get: jest.fn(async () => { return {}; }),
}));

jest.mock('../../../../src/account-manager/actions/AccountActions', () => ({
    asyncSaveAccountAction: jest.fn(async () => { }),
}));

jest.mock('../../../../src/account-manager/actions/NavigateActions', () => ({
    navigateForLink: jest.fn(),
}));

const createWebView = () => ({
    stopLoading: jest.fn(),
});

const createProps = (): ThisProps => {
    return {
        dispatch: jest.fn(),
        navigation: {
            pop: jest.fn(),
        } as any,
        state: {
            accountListState: {
                accounts: [],
                authState: {
                },
            },
            account: {
                authType: 'password',
                eimToken: [],
                id: 'site001',
                lastConnect: null,
                password: 'pass001',
                siteDomain: 'site-domain',
                siteName: 'site-name',
                userId: 'user001',
            },
        },
    };
};

describe('init', () => {
    test('auth type: password', () => {
        const wrapper = shallow<WebSignIn>(<WebSignIn {...createProps()} />);
        expect(toJson(wrapper)).toMatchSnapshot();
        const state = wrapper.state();
        expect(state.uriSource).toEqual({ uri: 'https://site-domain/services/login' })
    });
    test('auth type: o365', () => {
        const props = createProps();
        props.state.account!.authType = 'o365';
        const wrapper = shallow<WebSignIn>(<WebSignIn {...props} />);
        expect(toJson(wrapper)).toMatchSnapshot();
        const state = wrapper.state();
        expect(state.uriSource).toEqual({ uri: 'https://site-domain' })
    });
    test('no account', () => {
        const props = createProps();
        props.state.account = undefined;
        const wrapper = shallow<WebSignIn>(<WebSignIn {...props} />);
        expect(toJson(wrapper)).toMatchSnapshot();
        const state = wrapper.state();
        expect(state.uriSource).toEqual({ uri: 'https://' })
    });
});

describe('events', () => {
    let wrapper: Enzyme.ShallowWrapper<ThisProps, ILocalState, WebSignIn>;
    let instance: WebSignIn;
    let props: ThisProps;
    let account: IAccountState;
    beforeEach(() => {
        props = createProps();
        wrapper = shallow(<WebSignIn {...props} />);
        instance = wrapper.instance();
        account = props.state.account!;
    });
    test('onMessage', () => {
        instance['onMessage']({
            nativeEvent: {
                data: '{"userId": "user-id", "password": "pass-word"}'
            },
        } as any);
        const account = props.state.account!;
        account.userId = 'user-id';
        account.password = 'pass-word';
        expect(asyncSaveAccountAction).toBeCalledWith(account, props.dispatch);
    });
    test('set ref', () => {
        const dummy = createWebView();
        instance['setRef'](dummy as any);
        expect(instance['webview']).toEqual(dummy);
    });
    describe('postIdPassEimForm', () => {
        test('url is not applicable', () => {
            const result = instance['postIdPassEimForm']('https://exsample.com/', account);
            expect(result).toBeFalsy();
        });
        test('userid no set', () => {
            // ユーザー パスワードが指定されていない
            account.userId = undefined;
            expect(instance['postedIdPass']).toBeFalsy();
            const result = instance['postIdPassEimForm']('https://exsample.com/services/login', account);
            expect(result).toBeFalsy();
            expect(instance['postedIdPass']).toBeTruthy();
        });
        test('password id no set', () => {
            // パスワードが設定されていない
            account.password = undefined;
            expect(instance['postedIdPass']).toBeFalsy();
            const result = instance['postIdPassEimForm']('https://exsample.com/services/login', account);
            expect(result).toBeFalsy();
            expect(instance['postedIdPass']).toBeTruthy();
        });
        test('success', () => {
            account.userId = 'user%001';
            account.password = 'p@ssword';
            expect(instance['postedIdPass']).toBeFalsy();
            const result = instance['postIdPassEimForm']('https://exsample.com/services/login', account);
            expect(result).toBeTruthy();
            expect(instance['postedIdPass']).toBeTruthy();
            const state = wrapper.state();
            expect(state.uriSource).toEqual({
                body: 'userName=user%25001&password=p%40ssword',
                headers: {
                    'Content-Type': 'application/x-www-form-urlencoded;charset=UTF-8',
                },
                method: 'POST',
                uri: 'https://exsample.com/services/v1/login',
            });
            // 次回以降は実行されない
            const result2 = instance['postIdPassEimForm']('https://exsample.com/services/login', account);
            expect(result2).toBeFalsy();
        });
    });
    describe('postIdPassFor365', () => {
        //https://adfs.jp.ricoh.com/adfs/ls/
        test('url is not applicable', () => {
            const result = instance['postIdPassFor365'](
                'https://exsample.com/', account);
            expect(result).toBeFalsy();
        });
        test('userid no set', () => {
            expect(instance['postedIdPass']).toBeFalsy();
            account.userId = undefined;
            const result = instance['postIdPassFor365'](
                'https://adfs.jp.ricoh.com/adfs/ls/', account);
            expect(result).toBeFalsy();
            expect(instance['postedIdPass']).toBeTruthy();
        });
        test('password no set', () => {
            expect(instance['postedIdPass']).toBeFalsy();
            account.password = undefined;
            const result = instance['postIdPassFor365'](
                'https://adfs.jp.ricoh.com/adfs/ls/', account);
            expect(result).toBeFalsy();
            expect(instance['postedIdPass']).toBeTruthy();
        });
        test('success', () => {
            expect(instance['postedIdPass']).toBeFalsy();
            account.userId = 'user%001';
            account.password = 'p@ssword';
            const result = instance['postIdPassFor365'](
                'https://adfs.jp.ricoh.com/adfs/ls/', account);
            expect(result).toBeTruthy();
            expect(instance['postedIdPass']).toBeTruthy();
            const state = wrapper.state();
            expect(state.uriSource).toEqual({
                body: 'AuthMethod=FormsAuthentication' +
                    '&UserName=user%25001&Password=p%40ssword',
                headers: {
                    'Content-Type': 'application/x-www-form-urlencoded;charset=UTF-8',
                },
                method: 'POST',
                uri: 'https://adfs.jp.ricoh.com/adfs/ls/',
            });
            // 2回目は実施しない
            const result2 = instance['postIdPassFor365'](
                'https://adfs.jp.ricoh.com/adfs/ls/', account);
            expect(result2).toBeFalsy();
        });
    });
    describe('onLoadStartWebView', () => {
        beforeEach(() => {
            mocked(Cookies.get as jest.Mock).mockClear();
            mocked(Toast.show).mockClear();
            mocked(asyncSaveAccountAction).mockClear();
            instance['postIdPassEimForm'] = jest.fn().mockReturnValue(true);
            instance['postIdPassFor365'] = jest.fn().mockReturnValue(true);
            instance['webview'] = { stopLoading: jest.fn() } as any;
            wrapper.update();
        });
        test('no url', async () => {
            const dummyNavEvent = {
                nativeEvent: {
                    url: undefined
                },
            };
            await instance['onLoadStartWebView'](dummyNavEvent as any);
            expect(instance['postIdPassFor365']).not.toBeCalled();
            expect(instance['postIdPassEimForm']).not.toBeCalled();
            expect(instance['webview']!.stopLoading).not.toBeCalled();
        });

        test('auth o365', async () => {
            instance['saveProp']!.state.account!.authType = 'o365';
            wrapper.update();
            const dummyNavEvent = {
                nativeEvent: {
                    url: 'https://exsample.com/'
                },
            };
            await instance['onLoadStartWebView'](dummyNavEvent as any);
            expect(instance['postIdPassFor365']).toBeCalled();
            expect(instance['postIdPassEimForm']).not.toBeCalled();
            expect(instance['webview']!.stopLoading).toBeCalled();
        });

        test('auth password', async () => {
            instance['saveProp']!.state.account!.authType = 'password';
            wrapper.update();
            const dummyNavEvent = {
                nativeEvent: {
                    url: 'https://exsample.com/'
                },
            };
            await instance['onLoadStartWebView'](dummyNavEvent as any);
            expect(instance['postIdPassFor365']).not.toBeCalled();
            expect(instance['postIdPassEimForm']).toBeCalled();
            expect(instance['webview']!.stopLoading).toBeCalled();
        });

        test('no auto auth to success - password', async () => {
            instance['saveProp']!.state.account!.authType = 'password';
            instance['postIdPassEimForm'] = jest.fn().mockReturnValue(false);
            mocked(Cookies.get as jest.Mock).mockImplementation(async () => {
                return { APISID: 'got_token' };
            });
            wrapper.update();
            const dummyNavEvent = {
                nativeEvent: {
                    url: 'https://exsample.com/'
                },
            };
            await instance['onLoadStartWebView'](dummyNavEvent as any);
            expect(instance['postIdPassFor365']).not.toBeCalled();
            expect(instance['webview']!.stopLoading).toBeCalled();
            expect(Toast.show).toBeCalledWith({
                text: '認証に成功しました。',
                type: 'success',
            });
            const account = instance['saveProp']!.state.account!;
            expect(account.eimToken).toEqual(['APISID=got_token']);
            expect(asyncSaveAccountAction).toBeCalledWith(
                account, props.dispatch);
        });

        test('no auto auth to get not token - o365', async () => {
            instance['saveProp']!.state.account!.authType = 'o365';
            instance['postIdPassFor365'] = jest.fn().mockReturnValue(false);
            mocked(Cookies.get as jest.Mock).mockImplementation(async () => {
                return {};
            });
            wrapper.update();
            const dummyNavEvent = {
                nativeEvent: {
                    url: 'https://exsample.com/'
                },
            };
            await instance['onLoadStartWebView'](dummyNavEvent as any);
            expect(instance['postIdPassEimForm']).not.toBeCalled();
            expect(instance['webview']!.stopLoading).not.toBeCalled();
            expect(Toast.show).not.toBeCalled();
            expect(asyncSaveAccountAction).not.toBeCalled();
        });
    })
});


describe('navigationOptions', () => {
    test('', () => {
        const options = WebSignIn.navigationOptions();
        expect(options).toEqual({
            headerStyle: {
                backgroundColor: '#162756',
            },
            headerTintColor: '#fff',
            headerTitle: '',
        });
    })
});

describe('mapStateToProps', () => {
    test('', () => {
        const state = createProps().state;
        const result = __private__.mapStateToProps({
            webSignIn: { account: state.account },
            accountList: state.accountListState,
        } as any);
        expect(result).toStrictEqual({
            state,
        });
    });
});
