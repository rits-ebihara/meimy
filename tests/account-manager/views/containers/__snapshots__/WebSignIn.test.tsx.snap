// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`init auth type: o365 1`] = `
<Styled(Container)>
  <WebView
    cacheEnabled={true}
    injectedJavaScript="


(function(){
    var userNameInput = document.querySelector(\\"#userNameInput\\");
    var passwordInput = document.querySelector(\\"#passwordInput\\");
    var button = document.querySelector(\\"#submitButton\\");
    if (!button || !userNameInput || !passwordInput) { return; }
    button.addEventListener(\\"click\\", function() {
        var userPass = {userId: '', password: ''};
        userPass.userId = userNameInput.value;
        userPass.password = passwordInput.value;
        window.postMessage(JSON.stringify(userPass));
    });
})();
"
    onLoadStart={[Function]}
    onMessage={[Function]}
    originWhitelist={
      Array [
        "http://*",
        "https://*",
      ]
    }
    source={
      Object {
        "uri": "https://site-domain",
      }
    }
    useSharedProcessPool={true}
    useWebKit={true}
  />
</Styled(Container)>
`;

exports[`init auth type: password 1`] = `
<Styled(Container)>
  <WebView
    cacheEnabled={true}
    injectedJavaScript="

(function(){
    var eimSys = document.getElementById(\\"eim-system\\");
    if (!!eimSys) {
        eimSys.style.minWidth = \\"0\\";
    }
})();

(function(){
    var form = document.querySelector(\\"[name=form1]\\");
    if (!form) { return; }
    form.addEventListener(\\"submit\\", function() {
        var userNameInput = document.querySelector(\\"[name=userName]\\");
        var passwordInput = document.querySelector(\\"[name=password]\\");
        var userPass = {userId: '', password: ''};
        if (!!userNameInput) { userPass.userId = userNameInput.value; }
        if (!!passwordInput) { userPass.password = passwordInput.value; }
        window.postMessage(JSON.stringify(userPass));
    });
})();
"
    onLoadStart={[Function]}
    onMessage={[Function]}
    originWhitelist={
      Array [
        "http://*",
        "https://*",
      ]
    }
    source={
      Object {
        "uri": "https://site-domain/services/login",
      }
    }
    useSharedProcessPool={true}
    useWebKit={true}
  />
</Styled(Container)>
`;

exports[`init no account 1`] = `""`;
